design/vela-cli/sdk_generating.md

```diff
@@ -0,0 +1,114 @@
+# SDK Generating
+
+# Introduce
+
+This proposal introduces a framework which can generate KubeVela SDK from scaffolds, templates and X-Definitions.
+
+# Background
+
+As KubeVela designs for separation of concerns, there are two roles of user: platform team and end user. They are dealing with Application in different ways.
+
+![image](../resources/separation-of-roles.png)
+
+### For platform team
+
+When platform team users assembling Application, filling in components, traits and workflow steps, they are actually assemble Application YAML. Unfortunately, due to the extensibility of KubeVela X-Definition system, the property structure of different component (also for traits, workflow steps, policies) varies. In the source code, we use a `map[string]interface{}` to accept all kinds of properties. The unstructured code make it hard examine, modify, insert values of nested properties.
```

Comment:
the platform team don't assemble applications, they install addons, provide X-Definition for extension as their best practices.

Reply:
The platform team can hardly convince the end user to adopt KubeVela because they can't provide easy-to-use way to assemble Application through SDK without this proposal.

Reply:
> the platform team don't assemble applications

Yes, they don't. But they provide platform. By "platform team assembling applications", I mean they will build platform which accept argument from end user, then platform assemble the application automatically. This is what "platform" mean. But this kinds of usage will be easily covered by openapi schema and ui-schema like VelaUX has done.

> The platform team can hardly convince the end user to adopt KubeVela because they can't provide easy-to-use way to assemble Application through SDK without this proposal.

True.

Reply:
Your description here cause confusion, you'd better refine it.